nonthermal.cc

```diff
@@ -155,30 +158,114 @@ static struct nt_solution_struct *nt_solution;
 static double *deposition_rate_density;
 static int *deposition_rate_density_timestep;
 
-static void read_binding_energies() {
-  FILE *binding = fopen_required("binding_energies.txt", "r");
+static void read_shell_configs() {
+  auto shells_file = fstream_required("shells.txt", std::ios::in);
+
+  int nshells = 0; //number of shell in binding energy file
+  int n_z_binding = 0; //number of elements in file
 
-  int dum1 = 0;
-  int dum2 = 0;
-  assert_always(fscanf(binding, "%d %d", &dum1, &dum2) == 2);  // dimensions of the table
-  if ((dum1 != M_NT_SHELLS) || (dum2 != MAX_Z_BINDING)) {
+  std::string line;
+  assert_always(get_noncommentline(shells_file, line));
+  std::istringstream(line) >> nshells >> n_z_binding;
+  printout("Reading shells.txt with %d elements and %d shells\n", n_z_binding, nshells);
+
+  if ((nshells > M_NT_SHELLS) || (n_z_binding > MAX_Z_BINDING)) {
     printout("Wrong size for the binding energy tables!\n");
-    std::abort();
+    abort();
   }
 
-  for (int index1 = 0; index1 < dum2; index1++) {
-    float dum[10];
-    assert_always(fscanf(binding, "%g %g %g %g %g %g %g %g %g %g", &dum[0], &dum[1], &dum[2], &dum[3], &dum[4], &dum[5],
-                         &dum[6], &dum[7], &dum[8], &dum[9]) == 10);
+  int elementcounter = 0;
+  while (get_noncommentline(shells_file, line)) {
+    int q;
+    int z_element;
+    std::istringstream ssline(line);
 
-    for (int index2 = 0; index2 < 10; index2++) {
-      electron_binding[index1][index2] = dum[index2] * EV;
+    assert_always(ssline >> z_element);
+    printout("Reading shells Z=%d\n", z_element);
+
+    for (int shell = 0; shell < nshells; shell++){
+      assert_always(ssline >> q);
+      printout("q of %d in shell %d element number %d Z=%d\n", q, shell, elementcounter, z_element);
+      shells_q[elementcounter][shell] = q;
     }
+    elementcounter++;
   }
+}
+
+inline bool exists_test (const std::string& name) {
+  if (FILE *file = fopen(name.c_str(), "r")) {
+    fclose(file);
+    return true;
+  } else {
+    return false;
+  }
+}
```

Comment:
Is there a reason not to use std::filesystem::exists() here?

Reply:
Just didn't know about it. Feel free to replace with whatever you like